src/Setup.hs

```diff
@@ -4,17 +4,29 @@ import           Data.Char (isDigit, toLower)
 import           Data.Function (on)
 import           Data.List (intercalate, sortBy)
 import           Data.Monoid ((<>))
-import           Data.Version (showVersion)
 
-import           Distribution.InstalledPackageInfo
-import           Distribution.PackageDescription
-import           Distribution.Simple
+import           Distribution.InstalledPackageInfo (sourcePackageId)
+import           Distribution.PackageDescription (PackageDescription (..))
+import           Distribution.Simple (buildHook, defaultMainWithHooks, pkgName, pkgVersion, preConf, replHook
+                                     , sDistHook, simpleUserHooks, testHook, unPackageName)
+import           Distribution.Simple.BuildPaths (autogenPackageModulesDir)
 import           Distribution.Simple.Setup (BuildFlags(..), ReplFlags(..), TestFlags(..), fromFlag)
-import           Distribution.Simple.LocalBuildInfo
-import           Distribution.Simple.PackageIndex
-import           Distribution.Simple.BuildPaths (autogenModulesDir)
+import           Distribution.Simple.LocalBuildInfo (LocalBuildInfo, installedPkgs)
+import           Distribution.Simple.PackageIndex (allPackages)
 import           Distribution.Simple.Utils (createDirectoryIfMissingVerbose, rewriteFile, rawSystemStdout)
-import           Distribution.Verbosity
+import           Distribution.Verbosity (Verbosity, silent)
+
+import           Distribution.Version (Version, showVersion, versionNumbers)
+import           Distribution.Simple.BuildPaths (autogenModulesDir)
+
+-- This version of Setup.hs only builds against version 2.0 of the Cabal
+-- library, but can be built with the cabal-install version 1.24 as long
+-- as the cabal file includes:
+--
+--    custom-setup
+--        setup-depends:       base
+--                           , Cabal >= 2.0
```

Comment:
how does mafia cope with the `custom-setup` section? I know it's caused some problems in the past is all

Reply:
It copes as long as cabal-install is 1.24 or later. Previous problems were probably cabal-install 1.22 which I have confirmed not to work.